src/System.Private.Reflection.Core/src/System/Reflection/Runtime/Assemblies/EcmaFormat/EcmaFormatRuntimeAssembly.cs

```diff
@@ -97,22 +95,142 @@ public sealed override IEnumerable<Type> ExportedTypes
+                            PEReader pe = ModuleList.Instance.GetModuleInfoForMetadataReader(MetadataReader).EcmaPEInfo.PE;
+
+                            PEMemoryBlock resourceDirectory = pe.GetSectionData(pe.PEHeaders.CorHeader.ResourcesDirectory.RelativeVirtualAddress);
+                            BlobReader reader = resourceDirectory.GetReader((int)resource.Offset, resourceDirectory.Length - (int)resource.Offset);
+                            uint length = reader.ReadUInt32();
+                            result.PointerToResource = reader.CurrentPointer;
+                            
+                            // Length check the size of the resource to ensure it fits in the PE file section, note, this is only safe as its in a checked region
+                            if (length + sizeof(Int32) > reader.Length)
+                                throw new BadImageFormatException();
+                            result.SizeOfResource = length;
+                        }
+                    }
+                    else
+                    {
+                        if (resource.Implementation.Kind == HandleKind.AssemblyFile)
+                        {
+                            // Get file name
+                            result.ResourceLocation = default(ResourceLocation);
+                            AssemblyFile file = MetadataReader.GetAssemblyFile((AssemblyFileHandle)resource.Implementation);
+                            if (file.ContainsMetadata)
+                            {
+                                result.ResourceLocation = ResourceLocation.Embedded;
+                                throw new PlatformNotSupportedException(); // Support for multi-module assemblies is not implemented on this platform
+                            }
+                            result.FileName = MetadataReader.GetString(file.Name);
+                        }
+                        else if (resource.Implementation.Kind == HandleKind.AssemblyReference)
+                        {
+                            // Resolve assembly reference
+                            result.ResourceLocation = ResourceLocation.ContainedInAnotherAssembly;
+                            RuntimeAssemblyName destinationAssemblyName = ((AssemblyReferenceHandle)resource.Implementation).ToRuntimeAssemblyName(MetadataReader);
+                            result.ReferencedAssembly = RuntimeAssembly.GetRuntimeAssemblyIfExists(destinationAssemblyName);
+                        }
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public sealed override ManifestResourceInfo GetManifestResourceInfo(string resourceName)
         {
-            throw new NotImplementedException();
-            //return ReflectionCoreExecution.ExecutionEnvironment.GetManifestResourceInfo(this, resourceName);
+            if (resourceName == null)
+                throw new ArgumentNullException(nameof(resourceName));
+
```

Comment:
Nit: I think the code "paragraphs" better without the blank line between the two parameter validation checks.